### PR TITLE
Enhance example and add assets api

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log
- Enhance the sample code and add the Assets API.

### CODE REVIEW

**Comment by peter279k** [comment]
Hi @Josantonius, thank you for your reply.
I also let the different API be the different resources.
You can see more details about this [folder](https://github.com/PayCrypto/crypto-monitor/tree/master/src/Resource).

I think we have to separate the APIs for the different resources because each one resource has the different response JSON collection.

It's necessary to separate that and another reason is convenient to do the debug work.
Also, we can just focus on implementing the response JSON records.
If we implement all works in ```CryptoMonitor``` class, it's not maintainable and violates the single-responsibility principle of  developing classes.

**Comment by josantonius** [comment]
Hello Peter, how are you?

Sorry for the delay, between work and holidays I haven't had much time.

I wasn't referring specifically to the class structure, which is pretty good, but to the use that the end user will make of it.

Now we would:

```
use Joomla\DI\Container;
use ScriptFUSION\Porter\Porter;
use PayCrypto\CryptoMonitor;
use PayCrypto\Connector\CryptoConnector;
use PayCrypto\Resource\GetSpecificRate;
use PayCrypto\Resource\GetAllRate;
use PayCrypto\Resource\GetAssets;
use ScriptFUSION\Porter\Specification\ImportSpecification;

// Get specific rate

$apiKey = 'your-coin-api-key';

$container = new Container;
$container->set(CryptoMonitor::class, new CryptoMonitor(new CryptoConnector($apiKey)));
$porter = new Porter($container);
$specificRate = new GetSpecificRate($apiKey, 'BTC', 'USD');

$rates = $porter->importOne(new ImportSpecification($specificRate));

var_dump($rates);
```

I think we could do that internally and the end use would be much more simplified:

- Create an object to which we pass the provider and the API key.

- Use that object to get the rate.

Which would lead to something like that:

```
use PayCrypto\Foo;

$Foo = new Foo('provide-namer', 'your-coin-api-key'); // or using setters

$Foo->getSpecificRate('BTC', 'USD');
```

Basically the idea would be to create an extra class to take care of this (Container, Porter...) and this simplify the final use of the library.

What do you think?

**Comment by peter279k** [comment]
Hi @Josantonius, I glad that I see your comment again :smile:.
Yeah. Your suggestion looks good. How about overriding the `CryptoMonitor` class constructor to complete your recommendation?
```php
public function __construct(Connector $connector) { ... }
```
If possible, can you create the issue about this? Thanks.

**Comment by josantonius** [comment]
Yeah, give me a few days to catch up and I see how we do it  :wink:.

I also have the README file almost ready, but I stopped at the methods because it was a little confusing this way.

**Comment by josantonius** [comment]
Hi @peter279k,

I've uploaded my suggestion to the next [branch](https://github.com/PayCrypto/crypto-monitor/tree/crypto_monitor_josantonius).

It could be polished further, but the idea would be something like that.

**Comment by peter279k** [comment]
Hi @Josantonius, thank you for your reply.

I also create the realted PR #9.

Please review this code at your available time.
